display_list/display_list.cc

```diff
@@ -166,18 +255,25 @@ static bool CompareOps(uint8_t* ptrA,
   return true;
 }
 
-void DisplayList::RenderTo(DisplayListBuilder* builder,
-                           SkScalar opacity) const {
+void DisplayList::RenderTo(DisplayListBuilder* builder, SkScalar opacity) {
   // TODO(100983): Opacity is not respected and attributes are not reset.
   if (!builder) {
     return;
   }
-  Dispatch(*builder);
+  if (has_rtree()) {
+    Dispatch(*builder, builder->getLocalClipBounds());
+  } else {
+    Dispatch(*builder);
+  }
 }
 
-void DisplayList::RenderTo(SkCanvas* canvas, SkScalar opacity) const {
```

Comment:
nit: We can continue to mark these "Dispatch/RenderTo" methods const, if we mark `bounds()`, `rtree()`, `has_rtree()` as const

Reply:
Done. I made pretty much all fields and methods const.